orttraining/orttraining/python/training/ortmodule/ortmodule.py

```diff
@@ -24,8 +22,9 @@ class ORTModule(torch.nn.Module):
     """
 
     def __init__(self, module):
-        assert isinstance(
-            module, torch.nn.Module), "'module' must be a torch.nn.Module"
+        self._validate_module_type(module)
```

Comment:
Moving this to `GraphExecutionManager` allows users overriding this in their graph execution implementation without touching main `ORTModule` class

Reply:
The ```GraphExecutionManager``` is instantiated in ```ORTModule``` ```__init__```. If this were to move to ```GraphExecutionManager```, it wouldn't make a difference as of now.

I can move it there, but in order for the user to use ```ORTModule``` with an onnx model (for example), they would need to monkeypatch not only ```ORTModule```, but also the torch ```GraphExecutionManager```.

Reply:
By moving validation method to `GraphExecutionManager` the users will not monkey-patch validate, but they will implement it entirely on their implementation. say `OnnxInferenceManager.validate_module`

Replacing a factory is a temporary solution (and and exceptional case) until we decide to officially support ONNX or drop it. On a normal graph execution implementation, the factory would be extended with the new type. We are not doing this right now just because ONNX model is a tricky case that we are not sure about its future, but we need to support 1p.

Another possibility is moving this check to `GraphExecutionManager.__init__` and get rid of the extra method. The new inference/training implementation for ONNX would handle their validation - I think it is a bit more fragile than having a method on the class though

Reply:
Addressed.